include/up-cpp/client/usubscription/SubscriptionClient.h

```diff
@@ -0,0 +1,92 @@
+// SPDX-FileCopyrightText: 2024 Contributors to the Eclipse Foundation
+//
+// See the NOTICE file(s) distributed with this work for additional
+// information regarding copyright ownership.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Apache License Version 2.0 which is available at
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef UP_CPP_CLIENT_SubscriptionClient_H
+#define UP_CPP_CLIENT_SubscriptionClient_H
+
+#include <up-cpp/communication/Subscriber.h>
+#include <uprotocol/v1/umessage.pb.h>
+
+#include <optional>
+#include <uprotocol/core/usubscription/v3/usubscription.proto>
+#include <utility>
+
+namespace uprotocol::client::usubscription {
+/// @brief Interface for uEntities to create subscriptions.
+///
+/// Like all L3 client APIs, the SubscriptionClient is a wrapper on top of the
+/// L2 Communication APIs and USubscription service.
+struct SubscriptionClient {
+	using SubscriptionClientOrStatus =
+	    utils::Expected<std::unique_ptr<SubscriptionClient>, v1::UStatus>;
+	using ListenCallback = transport::UTransport::ListenCallback;
+	using ListenHandle = transport::UTransport::ListenHandle;
+	using SubscriptionResponse = core::usubscription::v3::SubscriptionResponse;
+
+	/// @brief Create a subscription
+	///
+	/// @param topic Topic to subscribe to.
+	/// @param callback Function that is called when a message is received.
+	/// @param ttl Time to live for the subscription.
+	/// @param sample_period_ms Sample period of messages for remote only
+	/// subscriptions.
+	/// @param subscription_details Additional details about the
+	/// subscription.
+	/// @param subscriber_details Additional details about the subscriber.
+	[[nodiscard]] SubscriptionClientOrStatus create(
+		std::shared_ptr<transport::UTransport> transport,
+	    const v1::UUri& topic, ListenCallback&& callback,
+	    std::optional<std::chrono::milliseconds> ttl = {},
+	    std::optional<uint32_t> sample_period_ms = {},
+	    std::optional<google::protobuf::Any>&& subscription_details = {},
+	    std::optional<google::protobuf::Any>&& subscriber_details = {});
+
+	/// @brief get subscription response
+	///
+	/// @returns SubscriptionResponse provided by uSubscription service
+	SubscriptionResponse getSubscriptionResponse() const {
+		return subscription_response_;
+	}
```

Comment:
I'm not sure if this is _really_ needed - we can probably handle the response internally and provide just a status outward. I'll need to think on it a little more.

Reply:
`message SubscriptionResponse {
  // Current status of the subscription
  SubscriptionStatus status = 1;

  // Any platform (uDevice) specific additional event delivered configuration
  // information for how the subscriber should consume the published events.
  EventDeliveryConfig config = 2;

  // Subscription topic passed to SubscriptionRequest
  uprotocol.v1.UUri topic = 3;
}`

As SubscriptionResponse may contain configuration information  which might be critically to consume information